packages/core/src/odata/common/uri-conversion/get-filter.ts

```diff
@@ -130,6 +134,42 @@ export function createGetFilter(uriConverter: UriConverter) {
         uriConverter.convertToUriFormat(filter.value, filter.edmType!)
       ].join(' ');
     }
+
+    if (isFilterLambdaExpression(filter)) {
+      return getODataFilterExpressionWhenBeingFilterLambdaExpression(
+        filter,
+        parentFieldNames,
+        targetEntityConstructor
+      );
+    }
+  }
+
+  function getODataFilterExpressionWhenBeingFilterLambdaExpression<
```

Comment:
I extracted it in this way, because I thought all the different path (e.g., when isFilter() returns true) in the `getODataFilterExpression` should be moved to the target class (Filter class).
So I created this function and even for all different classes.
However, this will result in additional circular dependency issue.
So I gave up, since I don't find a simple solution and don't want to spend too much time.
As a result, only this new function is there.